src/oneWay.ts

```diff
@@ -0,0 +1,92 @@
+// @ts-ignore
+import arrayMean from 'ml-array-mean';
+// @ts-ignore
+import * as cephes from 'cephes';
+
+export interface IOneWayOptions {
+  alpha?: number;
+}
+
+export interface IOneWayResult {
+  rejected: boolean;
+  testValue: number;
+  pValue: number;
+  freedom: [number, number];
+}
+
+export function oneWay(
+  data: number[],
+  classes: unknown[],
+  options: IOneWayOptions = {},
+): IOneWayResult {
+  if (!Array.isArray(data)) {
+    throw new TypeError('data must be an array');
+  }
+  if (!Array.isArray(classes)) {
+    throw new TypeError('classes must be an array');
+  }
+  if (data.length !== classes.length) {
+    throw new RangeError('data must have the same length as classes');
+  }
+
+  const { alpha = 0.05 } = options;
+
+  const allClasses = new Set(classes);
+  const numClasses = allClasses.size;
+
+  if (numClasses < 2) {
+    throw new RangeError('there must be at least two different classes');
+  }
+
+  const classToIndex = new Map<unknown, number>();
+  {
+    let i = 0;
+    for (const klass of allClasses) {
+      classToIndex.set(klass, i++);
+    }
+  }
+
+  const d1 = numClasses - 1;
+  const d2 = data.length - numClasses;
+
+  const allData: number[][] = new Array(numClasses);
+  for (let i = 0; i < numClasses; i++) {
+    allData[i] = [];
+  }
+
+  for (let i = 0; i < data.length; i++) {
+    const index = classToIndex.get(classes[i]) as number;
```

Comment:
why do you need as number? Isn't the map supposed to return a number already?

Reply:
it can return `undefined` if the key was not found, but here we are sure it won't be the case so I cast it.